File: tests/testdata/codegen/components-pp/go/components.go
package main

import (
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		_, err := NewSimpleComponent(ctx, "simpleComponent", nil)
		if err != nil {
			return err
		}
		var multipleSimpleComponents []*SimpleComponent
		for index := 0; index < 10; index++ {
			key0 := index
			_ := index
			__res, err := NewSimpleComponent(ctx, fmt.Sprintf("multipleSimpleComponents-%v", key0), nil)
			if err != nil {
				return err
			}
			multipleSimpleComponents = append(multipleSimpleComponents, __res)
		}
		_, err = NewAnotherComponent(ctx, "anotherComponent", nil)
		if err != nil {
			return err
		}
		exampleComponent, err := NewExampleComponent(ctx, "exampleComponent", &ExampleComponentArgs{
			Input: "doggo",
			IpAddress: pulumi.IntArray{
				pulumi.Int(127),
				pulumi.Int(0),
				pulumi.Int(0),
				pulumi.Int(1),
			},
			CidrBlocks: map[string]interface{}{
				"one": "uno",
				"two": "dos",
			},
			GithubApp: &GithubAppArgs{
				Id:            "example id",
				KeyBase64:     "base64 encoded key",
				WebhookSecret: "very important secret",
			},
			Servers: []map[string]interface{}{
				&ServersArgs{
					Name: "First",
				},
				&ServersArgs{
					Name: "Second",
				},
			},
			DeploymentZones: map[string]interface{}{
				"first": &DeploymentZonesArgs{
					Zone: "First zone",
				},
				"second": &DeploymentZonesArgs{
					Zone: "Second zone",
				},
			},
		})
		if err != nil {
			return err
		}
		ctx.Export("result", exampleComponent.Result)
		return nil
	})
}

File: tests/testdata/codegen/components-pp/dotnet/components.cs
using System.Collections.Generic;
using System.Linq;
using Pulumi;

return await Deployment.RunAsync(() => 
{
    var simpleComponent = new Components.SimpleComponent("simpleComponent");

    var multipleSimpleComponents = new List<Components.SimpleComponent>();
    for (var rangeIndex = 0; rangeIndex < 10; rangeIndex++)
    {
        var range = new { Value = rangeIndex };
        multipleSimpleComponents.Add(new Components.SimpleComponent($"multipleSimpleComponents-{range.Value}"));
    }
    var anotherComponent = new Components.AnotherComponent("anotherComponent");

    var exampleComponent = new Components.ExampleComponent("exampleComponent", new()
    {
        Input = "doggo",
        IpAddress = new[]
        {
            127,
            0,
            0,
            1,
        },
        CidrBlocks = 
        {
            { "one", "uno" },
            { "two", "dos" },
        },
        GithubApp = new Components.ExampleComponentArgs.GithubAppArgs
        {
            Id = "example id",
            KeyBase64 = "base64 encoded key",
            WebhookSecret = "very important secret",
        },
        Servers = new[]
        {
            new Components.ExampleComponentArgs.ServersArgs
            {
                Name = "First",
            },
            new Components.ExampleComponentArgs.ServersArgs
            {
                Name = "Second",
            },
        },
        DeploymentZones = 
        {
            { "first", new Components.ExampleComponentArgs.DeploymentZonesArgs
            {
                Zone = "First zone",
            } },
            { "second", new Components.ExampleComponentArgs.DeploymentZonesArgs
            {
                Zone = "Second zone",
            } },
        },
    });

    return new Dictionary<string, object?>
    {
        ["result"] = exampleComponent.Result,
    };
});

